src/cdk-experimental/listbox/listbox.ts

```diff
@@ -7,98 +7,206 @@
- * Typically CdkOption elements are placed inside the listbox. Logic to handle keyboard navigation,
- * selection of options, active options, and disabled states is built in.
- */
 @Directive({
-  selector: '[cdkListbox]',
-  exportAs: 'cdkListbox',
-  host: {
-    role: 'listbox',
-  }
+    selector: '[cdkListbox]',
+    exportAs: 'cdkListbox',
+    host: {
+      role: 'listbox',
+      '(keydown)': '_keydown($event)',
+      '[attr.aria-disabled]': '_disabled',
+    }
 })
-export class CdkListbox {
+export class CdkListbox implements AfterContentInit, OnDestroy {
+
+  _listKeyManager: ActiveDescendantKeyManager<CdkOption>;
+  private _disabled: boolean = false;
 
-  /** A query list containing all CdkOption elements within this listbox */
   @ContentChildren(CdkOption, {descendants: true}) _options: QueryList<CdkOption>;
 
   @Output() readonly selectionChange: EventEmitter<ListboxSelectionChangeEvent> =
       new EventEmitter<ListboxSelectionChangeEvent>();
 
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+
+  ngAfterContentInit() {
+    this._listKeyManager = new ActiveDescendantKeyManager(this._options)
+      .withWrap().withVerticalOrientation().withTypeAhead();
+  }
+
+  ngOnDestroy() {
+    this._listKeyManager.change.complete();
+  }
+
+  _keydown(event: KeyboardEvent) {
+    if (this._disabled) {
+      return;
+    }
+
+    const keyCode = event.keyCode;
+
+    switch (keyCode) {
+      case SPACE:
+      case ENTER:
+        if (this._listKeyManager.activeItem && !this._listKeyManager.isTyping()) {
+          this._toggleActiveOption();
+        }
+        break;
+      default:
+        this._listKeyManager.onKeydown(event);
+    }
+  }
+
   /** Emits a selection change event, called when an option has its selected state changed */
   _emitChangeEvent(option: CdkOption) {
     this.selectionChange.emit(new ListboxSelectionChangeEvent(this, option));
   }
 
-  /** Sets the given option's selected state to true */
+  private _toggleActiveOption() {
+    const currentActiveOption = this._listKeyManager.activeItem;
+    if (currentActiveOption && !currentActiveOption.disabled) {
+      currentActiveOption.toggle();
+      this._emitChangeEvent(currentActiveOption);
+    }
+  }
+
   select(option: CdkOption) {
-    option.selected = true;
+    if (!this.disabled && option.disabled) {
+      option.selected = true;
+    }
   }
 
-  /** Sets the given option's selected state to null. Null is preferable for screen readers */
   deselect(option: CdkOption) {
-    option.selected = false;
+    if (!this.disabled && option.disabled) {
+      option.selected = false;
+    }
   }
+
+  setActiveOption(option: CdkOption) {
+    this._listKeyManager.updateActiveItem(option);
+  }
+
+  setDisabledOption(isDisabled: boolean, option: CdkOption) {
```

Comment:
I don't follow why this method is here- it doesn't seem to involve the listbox; the caller could just directly set `option.disabled`

Reply:
That is true especially since one parameter is the option already.